app/models/batch.rb

```diff
@@ -0,0 +1,36 @@
+class Batch < ApplicationRecord
+  has_and_belongs_to_many :checks
+
+  def to_h
+    {
+      id: id,
+      status: status,
+      links: checks.map(&:to_h),
+      totals: {
+        links: checks.count,
+        ok: checks.each.count { |check| check.status == "ok" },
+        caution: checks.each.count { |check| check.status == "caution" },
+        broken: checks.each.count { |check| check.status == "broken" },
+        pending: checks.each.count { |check| check.status == "pending" },
+      },
+      completed_at: completed_at,
+    }.deep_symbolize_keys
+  end
```

Comment:
I think this belongs in a presenter class, probably of name `BatchReport` so we can be consistent with terminology (hopefully we don't already have a collision on LinkReport 😬 )

I'd suggest that `to_h` is more of a place to serialize the model in the same way as `as_json` whereas we're presenting this as a domain concept.